src/prefect/blocks/core.py

```diff
@@ -120,6 +121,8 @@ class Block(BaseModel, ABC):
     class Config:
         extra = "allow"
 
+        json_encoders = {SecretDict: lambda v: v.dict()}
```

Comment:
Do we need to get_secret_values here since the value gets irreversibly obfuscated?

Reply:
No, this is to make sure that you can't see the values when `.json()` is called on a block the values aren't printed. Getting the secret values to send to the API happens when we create a block document with `_to_block_document`

Reply:
Yeah this seems like a JSON roundtrip would fail?

Reply:
I feel like we should only obfuscate values in `str` and `repr` but 🤷‍♀️

Reply:
The JSON payload comes from the `BlockDocument` model and not from the `Block` subclass directly. It might be too aggressive to obfuscate in `.json()` as well, but I thought it was the most expected behavior.

Reply:
Let's revisit this if/when it's a problem :)